cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -757,6 +760,112 @@ func (s *Server) handleDebugList(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) handleDebugDelete(w http.ResponseWriter, r *http.Request) {
+	rawID := r.URL.Query().Get("id")
+	if rawID == "" {
+		http.Error(w, "URL parameter 'id' must be specified", http.StatusBadRequest)
+		return
+	}
+
+	id64, err := strconv.ParseUint(rawID, 10, 32)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to parse repository id %q as uint32: %s", rawID, err), http.StatusBadRequest)
+		return
+	}
+
+	repoID := uint32(id64)
+
+	s.queue.mu.Lock()
+	defer s.queue.mu.Unlock()
```

Comment:
you are holding the queue lock while deleting. looks like you should only be holding it for the s.queue.get operation?